src/isomorphic/classic/element/ReactElement.js

```diff
@@ -284,14 +299,20 @@ ReactElement.cloneElement = function(element, config, children) {
         'Properties defined in its prototype chain will be ignored.'
       );
     }
-    if (config.ref !== undefined) {
+
+    var configRef = getRef(config);
+    var configKey = getKey(config);
+
+    if (configRef != null) {
       // Silently steal the ref from the parent.
```

Comment:
I think this approach should keep behavior consistent. Let me know what you think and I'll add a test for the `null` cases.

```js
function isValidConfigRefOrKey(config, name) {
  if (__DEV__) {
    return config.hasOwnProperty(name) &&
      !Object.getOwnPropertyDescriptor(config, name).get;
  }

  return config[name] !== undefined;
}

function getConfigKey(config) {
  return '' + config.key;
}

...

ReactElement.createElement = function(type, config, children) {
  ...

    if (isValidConfigRefOrKey(config, 'ref')) {
      ref = config.ref;
    }

    if (isValidConfigRefOrKey(config, 'key')) {
      key = getConfigKey(config);
    }

...

ReactElement.cloneElement = function(element, config, children) {
  ...

    if (isValidConfigRefOrKey(config, 'ref')) {
      // Silently steal the ref from the parent.
      ref = config.ref;
      owner = ReactCurrentOwner.current;
    }

    if (isValidConfigRefOrKey(config, 'key')) {
      key = getConfigKey(config);
    }
```